core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -62,32 +62,45 @@ public void inboundMessage() {
   }
 
   /**
+   * An outbound message has been completely serialized, revealing its wire size.
+   */
+  public void outboundMessageSerialized(long wireSize) {
```

Comment:
I think what option we choose we should be consistent on both cases. Which one do you think it is easier.

I will also have to consider maybe changing the NetworkEvent to support 2 sizes (compressed and uncompressed) to make it clear to the users which one it is.

I am not sure I understand all the options, but I would say consistency is important (between inbound and outbound and between languages), so I would like to know what is the story in other languages (Go, C)